trl/trainer/grpo_trainer.py

```diff
@@ -1217,29 +1218,31 @@ def _generate_and_score_completions(
                     unwrapped_model.to(torch.float16)
                 with torch.inference_mode():
                     all_outputs = unwrapped_model.generate_batch(
-                        prompt_inputs.input_ids, generation_config=self.generation_config
+                        prompt_inputs.input_ids, generation_config=self.generation_config, progress_bar=False
                     )
             completion_ids = [output.generated_tokens for output in all_outputs.values()]
             completion_ids = [torch.tensor(ids, device=device) for ids in completion_ids]
-            completion_ids = pad(completion_ids, padding_value=self.processing_class.pad_token_id)
+            completion_ids = pad(
+                completion_ids, padding_value=self.processing_class.pad_token_id, padding_side="right"
+            )
             prompt_ids = [torch.tensor(ids, device=device) for ids in prompt_inputs.input_ids]
-            prompt_ids = pad(prompt_ids, padding_value=self.processing_class.pad_token_id)
+            prompt_ids = pad(prompt_ids, padding_value=self.processing_class.pad_token_id, padding_side="left")
```

Comment:
Padding side must be `"left"` for prompts, otherwise you'd get (incorrect):

```
[[P01, P01, P03, PAD, PAD, C01, C02, C03, C04],
 [P11, P11, P13, P14, P15, C11, C12, PAD, PAD]]
```

instead of (correct)

```
[[PAD, PAD, P01, P01, P03, C01, C02, C03, C04],
 [P11, P11, P13, P14, P15, C11, C12, PAD, PAD]]
```

Reply:
the error is only when using CB, introduced in #3471